### ISSUE TITLE
Default settings/ empty peak files

### ISSUE THREAD

**Issue by sthair** [databio/pepatac#90]
Hi there, 
Can you point me to where your default settings are for each step, eg. alignment- is it bowtie? What settings? --sensitive or --very sensitive? trimming = 50bp overlap? MACS2 sliding window default?  Also, if I want to change these, can you give an example of the usage? Your usage page only explains the basics. 
(Forgive me if this is here somewhere but I can't find it and I've read several of the .py and .yaml scripts with limited insight). 
Also, sorry to post on github but your webform for 'contact us' fails (404 error).
Thanks!!
Simone

**Comment by vreuter** [comment]
Hi @sthair , you can tweak tool settings with the [pipeline configuration file](https://github.com/databio/pepatac/blob/master/pipelines/pepatac.yaml)

**Comment by sthair** [comment]
Thank you that is super helpful.  Still tweaking installation of the pipeline with help of cluster master.. ran the test example.. under example tutorial summary, if I click on Dedup_alignment_rate, the bar is at 1.75M. however ```samtools view -c -F 4 test1_sort_dedup.bam``` == 17728. same for test1_sort.bam. Because test1_temp.bam is not retained that is as far back as I could trace.. does that make sense? I am asking because the real problem is that ```cat test1_shift.bed | awk '{print $5}' | sort | uniq -c
  17728 0```- all zeros? and hence all narrowPeak files are empty.. I've gone through all of the lines that print while it runs and I cannot seem to determine what has happened.

**Comment by vreuter** [comment]
Hm I'm not sure right off hand, but this sounds sort of familiar so may've previously come up;  a scan through the closed issues didn't yield anything promising-looking to me, though. @jpsmith5 may know more or have a stronger intuition for what's at play. Regardless, if you still have a log file that was produced or could copy the printouts into a text file that you can attach, that'd be great and may help us figure it out.

**Comment by sthair** [comment]
OK I am running it again, module has been rebuilt and I will see whatI get

**Comment by sthair** [comment]
Hi there, @vreuter @jpsmith5 

OK I've run it again and same thing. 
I think the issue lies somewhere in the process of converting the bam to bed files. ??? Here is what I have gathered:

Aside from this one kinda cryptic message at the end of the macs2 step ```> `Peak_count`  0       PEPATAC _RES_```
there are no warnings or 'stopifs'
[export4GitHubTrouble.zip](https://github.com/databio/pepatac/files/3105379/export4GitHubTrouble.zip)
 until you get to ```> `Rscript /pepatac/tools/PEPATAC_frip.R test1 /test1/peak_calling_hg38/test1_peaks_coverage.bed 17728 /test1/QC_hg38/test1_frif.pdf --bed etctera etcetera.......` (40199)
<pre>
Peak file is empty```

It would appear (?) that test1_sort_dedup.bam has some data in it but whatever it is, test1_shift.bed is all zeros, which if I am reading the script correctly is the key file for calling peaks and all of the QC that follows. 
```samtools view -c -F 4 aligned_hg38/test1_sort_dedup.bam```
17728
```cat aligned_hg38_exact/test1_shift.bed | awk '{print $5}' | sort | uniq -c ```
  17728 0

Indeed, all of the peak files are empty
[sthair@sXXXX peak_calling_hg38]$ ls -lh
total 42K
-rw-r--r-- 1 sthair   0 Apr 22 16:02 test1_peaks.bigBed
-rw-r--r-- 1 sthair    0 Apr 22 16:02 test1_peaks_coverage.bed
-rw-r--r-- 1 sthair    0 Apr 22 16:02 test1_peaks.narrowPeak
-rw-r--r-- 1 sthair  1.1K Apr 22 16:02 test1_peaks.xls
-rw-r--r-- 1 sthair     0 Apr 22 16:02 test1_summits.bed

I've attached a couple files. Let me know if you need more info. 
Thanks
Simone

**Comment by sthair** [comment]
Good morning, 

FYI you may find in the log attached in my previous post ```Skipping TSS -- TSS enrichment requires TSS annotation file```. 

That has been corrected and run again, but the subsequent output still has empty peak files, which makes sense from what I glean from the code as the generation of those plots are a branch and I don't think they feed into the bam to bed steps. 

Nevertheless, here are the TSS and ISS plots from this latest run, 
[test1_fragLenDistribution.pdf](https://github.com/databio/pepatac/files/3108822/test1_fragLenDistribution.pdf)
[test1_TssEnrichment.pdf](https://github.com/databio/pepatac/files/3108825/test1_TssEnrichment.pdf)

In comparison, if I click on the TSS_score plot from here: 
http://code.databio.org/PEPATAC/files/examples/tutorial/tutorial_summary.html

The bar graph is sitting at 13.32. But I don't get what the bar graph represents exactly? Is the same as the value printed in the top right of the plot?  The peak of my TSS plot here is 8.5. 

The ISS plot is pretty jagged but I gather this is because we are working with a subset of data to keep the tests running quickly. (??)

Maybe quite rudimentary for you all but I've added since for me it confirms that we have data in the bam files, and hopefully you can compare with your expected plots.... to determine if data is missing from the bam files? 
if this is what they should look like then I feel a smidge more confident that the issue is still in the bamtobed step somewhere. 


Looking forward to hearing your thoughts! 
Thank you!

Cheers,
Simone

**Comment by sthair** [comment]
Hi-
So.... I got a hold of someone at Stanford familiar with your work as I was desperate to get this running who mentioned the test data ```examples/data/test1_r1.fastq.gz``` ```examples/data/test1_r2.fastq.gz``` are so small there may be no peaks to call. Because I was trading code with our server master who has to set up the container environments, I didn't realize that I should be comparing the ```examples/data/tutorial1_r1.fastq.gz``` ```examples/data/tutorial1_r2.fastq.gz``` to what is on your website.....**not test1.**......  Once I did that, everything matches up. 

Pretty simple explanation. 

Thanks
Simone

**Comment by sthair** [comment]
PS. I have deleted the posts above so the logs etc are not out there with my username etc. If you haven't got them and want them just ping me.

**Comment by nsheff** [comment]
Hi Simone,

Yeah, the test data is just to make sure everything can *run*...it's tiny. We didn't want to include any real data inside the code repository.

So where exactly was the confusion? Is there a point in the documentation we could fix to make it easier for future searchers?

Also if you're at Stanford, we already have pepatac running on Sherlock with all the compute settings set up: https://github.com/pepkit/pepenv/blob/master/stanford_sherlock.yaml

maybe that's useful?

**Comment by sthair** [comment]
Hi Nathan @nsheff !
Thank you so much for getting back to me. The confusion literally was a product of having too many deadlines all at the same time last week and trying to complete this project "in my spare time- aka the middle of the night". At the beginning I had to hand this off for the cluster master to get all of the programs installed/ get container working. 

I didn't read the full tutorial before he kindly sent me commands to try and hence I didn't realize test != tutorial as I had not finished reading the tutorial. (grimacing face)
 
I am not on Sherlock hence we needed to set up and test.. 

We have it running great now with hg_38 but... 
I am having issues running hg_19 alignments... which I need to use to align with previously analyzed data. I will make a separate comment for that so it doesn't get buried in unrelated comments. 
Thanks again
Simone
